src/index.ts

```diff
@@ -0,0 +1,516 @@
+        }
+      });
+
+      if (localCandidate && remoteCandidate) {
+        event.candidatePair = localCandidate.candidateType + ';' + remoteCandidate.candidateType;
+        event.candidatePairDetails = {
+          local: localCandidate,
+          remote: remoteCandidate,
+          pair: activeCandidatePair
+        };
+      }
+
+      if (localCandidate) {
+        event.transport = localCandidate.transport || localCandidate.protocol;
+        if (localCandidate.priority) {
+          // Chrome-specific mapping;
+          // but only chrome has priority set on the candidate currently.
+          const turnTypes = {
+            2: 'udp',
+            1: 'tcp',
+            0: 'tls'
+          };
+
+          const priority = parseInt(localCandidate.priority, 10);
+          event.turnType = turnTypes[priority >> 24];
+          event.networkType = localCandidate.networkType;
+        }
+
+        event.usingIPv6 = localCandidate.ipAddress && localCandidate.ipAddress.indexOf('[') === 0;
+      }
+    }
+  }
+
+  private async handleConnectionStateChange () {
+    const state = this.peerConnection.connectionState;
+
+    if (state === 'connected') {
+      this.pollForStats();
+    } else if (state === 'disconnected') {
+      if (this.peerConnection.signalingState !== 'stable') {
+        return;
+      }
+
+      return this.gatherStats().then((reports) => {
+        const event = this.createStatsReport(reports);
+        event.type = 'disconnected';
+        this.emit('stats', event);
+      });
+    } else if (state === 'closed') {
+      if (this.pollingInterval) {
+        if (IS_BROWSER) {
+          window.clearInterval(this.pollingInterval);
+        }
+        this.pollingInterval = null;
+      }
+    }
+  }
+
+  private pollForStats () {
+    if (this.pollingInterval) {
+      return;
+    }
+
+    const statsPoll = () => {
+      return this.gatherStats().then((reports) => {
+        const event = this.createStatsReport(reports, true);
+        if (event.tracks.length > 0 || event.remoteTracks.length > 0) {
+          this.emit('stats', event);
+        }
+      });
+    };
+
+    if (IS_BROWSER) {
+      window.setTimeout(statsPoll, 0);
+      this.pollingInterval = window.setInterval(statsPoll, this.statsInterval);
+    }
+  }
+
+  private polyFillStats (results: RTCStatsReport): Array<{key: RTCStatsType, value: any}> {
+    if (!results) {
+      return [];
+    }
+
+    if (Array.isArray(results)) {
+      return results;
+    }
+
+    const betterResults = [];
+
+    if (this.isNativeStatsReport(results)) {
+      results.forEach((value, key) => {
+        betterResults.push({ key, value });
+      });
+    } else if (Object.keys(results).length > 0) {
+      Object.keys(results).forEach(key => {
+        betterResults.push({
+          key,
+          value: results[key]
+        });
+      });
+    } else {
+      this.logger.warn('Unknown stats results format, returning unmodified', results);
```

Comment:
This log doesn't seem to make sense because it isn't actually returning the results unmodified. It is just returning an empty array.